Modules/LearningSequence/classes/class.ilObjLearningSequenceAdminGUI.php

```diff
@@ -0,0 +1,133 @@
+        }
+    }
+
+    public function executeCommand()
+    {
+        $this->checkPermission('read');
+        $next_class = $this->ctrl->getNextClass($this);
+        $cmd = $this->ctrl->getCmd();
+        $this->prepareOutput();
+        
+        switch ($next_class) {
+            case 'ilpermissiongui':
+                $this->tabs_gui->setTabActive('perm_settings');
+                $perm_gui = new ilPermissionGUI($this);
+                $this->ctrl->forwardCommand($perm_gui);
+                break;
+
+            default:
+                switch ($cmd) {
+                    case self::CMD_VIEW:
+                    case self::CMD_EDIT:
+                        $this->edit();
+                        break;
+                    case self::CMD_SAVE:
+                        $this->save();
+                        break;
+                    default:
+                        throw new Exception(__METHOD__ . " :: Unknown command " . $cmd);
+                }
+        }
+    }
+
+    protected function getForm(array $values = []) : Input\Container\Form\Form
+    {
+        $target = $this->ctrl->getLinkTargetByClass(self::class, self::CMD_SAVE);
+        $poll_interval = $this->ui_factory->input()->field()->numeric(
+            $this->lng->txt("lso_admin_interval_label"),
+            $this->lng->txt("lso_admin_interval_byline")
+        )
+        ->withAdditionalConstraint($this->validation_factory->greaterThan(0));
+
+        if (isset($values[self::F_POLL_INTERVAL])) {
+            $poll_interval = $poll_interval->withValue($values[self::F_POLL_INTERVAL]);
+        }
+
+        $section = $this->ui_factory->input()->field()->section(
+            [self::F_POLL_INTERVAL => $poll_interval],
+            $this->lng->txt("lso_admin_form_title"),
+            $this->lng->txt("lso_admin_form_byline")
+        );
+        $form = $this->ui_factory->input()->container()->form()->standard($target, [$section]);
+        return $form;
+    }
+
+    protected function show(Input\Container\Form\Form $form) : void
+    {
+        $this->tpl->setContent(
+            $this->ui_renderer->render($form)
+        );
+    }
+
+    protected function getCurrentPollingInterval() : float
+    {
+        $interval = $this->ilSetting->get(\ilObjLearningSequenceAdmin::SETTING_POLL_INTERVAL);
+        if (!$interval) {
+            $interval = \ilObjLearningSequenceAdmin::POLL_INTERVAL_DEFAULT;
+        }
+        return (float) $interval;
+    }
+
+    protected function edit() : void
+    {
+        $values = [
+            self::F_POLL_INTERVAL => $this->getCurrentPollingInterval()
+        ];
+        $form = $this->getForm($values);
+        $this->show($form);
+    }
+
+    protected function save() : void
+    {
+        $form = $this->getForm()->withRequest($this->request);
+        $data = $form->getData();
+        if ($data) {
+            $data = array_shift($data);
```

Comment:
I think this rather belongs to the form, because it is a detail on how the form is structured internally. So we should attach it to the form via an according transformation.